Tests/AESGCMSpecTest.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Yan Peng
+-/
+import Arm.BitVec
+import Specs.AES
+import Specs.GCM
+
+-- Reference: https://csrc.nist.rip/groups/ST/toolkit/BCM/documents/proposedmodes/gcm/gcm-spec.pdf
```

Comment:
It seems that this spec contains 18 test cases. Is there a rationale how 11 cases are chosen?

Reply:
Great observation. There are 12 test vectors (for AES-128-GCM and AES-256-GCM). The 6 left are for AES-192-GCM. I didn't add it because we are not interested in AES-192-GCM, but there is no harm.

Reply:
Done adding all tests for AES-192-GCM